### DISCUSSION TITLE
[@types/yargs] - ESM version is missing important typings

### DISCUSSION THREAD

**Discussion by Arilas** [DefinitelyTyped/DefinitelyTyped#63348] [Issues with a @types package]
`index.d.mts` is missing re-export of `ArgumentsCamelCase` type from `index.d.ts`, which is used when defining the command:

```ts
        command<U = T>(
            command: string | ReadonlyArray<string>,
            showInHelp: false,
            builder?: BuilderCallback<T, U>,
            handler?: (args: ArgumentsCamelCase<U>) => void | Promise<void>,
            middlewares?: MiddlewareFunction[],
            deprecated?: boolean | string,
        ): Argv<T>;
```

Without this, it's impossible to correctly type handler without ts-ignore.

This is reproducible with following tsconfig parameters:
```json
{
    "module": "Node16",
    "moduleResolution": "Node16",
}
```

[a comment by an automated account is not shown]

**Comment by tpluscode** [comment]
Is it the same as https://github.com/microsoft/TypeScript/issues/53349?

If so, then the types package may need `"type": "module"` in its `package.json`

**Reply by tpluscode** [reply]
Oh, this one is more difficult because the default exports are CJS and the package comes with ESM.

Maybe this is a problem in TS after all and not in type packages

**Comment by shadowspawn** [comment]
Related: #61189
